src/components/AssetManagementPanel/AssetManagementForm/FormVariants/SurrenderForm/SurrenderForm.tsx

```diff
@@ -35,6 +40,17 @@ export const SurrenderForm = ({
     onSetFormAction(AssetManagementActions.None);
   };
 
+  const { setOverlayContent } = useContext(OverlayContext);
+  const showDocumentSurrenderSuccess = useCallback(() => {
```

Comment:
I'm thinking that rather than using a callback can the userEffect check that

if(isConfirmed && callback's content is not the SURRENDER_DOCUMENT_SUCCESS) showDocumentSurrenderSuccess

This way we dont have to use an extra callback. It's not easy to understand what you are trying to do with the useCallback in the current way.